src/libstd/ffi/c_str.rs

```diff
@@ -969,9 +968,7 @@ impl CStr {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
A tangent but, since you were talking about checking in from_ptr, I was looking at the doc here and it could be more clear. It's written to list things that are "scary", but stops short of actually telling the user what to do. Do you see it the same way? It's like *"someone might pass invalid pointers"* (and then what?) instead of *"you must pass a valid pointer"*. :slightly_smiling_face: 

The doc should ideally clarify for the user that the pointer must be non-null and other important things. I'm not saying we need to write an exhaustive spec.

Reply:
I rewrote the unsafe docs to be more helpful

Reply:
Awesome :heart: